### ISSUE TITLE
Implement string.Trim with parameters on SQL Server

### ISSUE THREAD

**Issue by roji** [dotnet/efcore#22927]
We currently translate string.Trim only when no characters are specified (whitespace only), and implement it with LTRIM/RTRIM.

SQL Server 2017 introduced a more fully-featured [TRIM](https://docs.microsoft.com/en-us/sql/t-sql/functions/trim-transact-sql?view=sql-server-ver15) function which can accept characters to trim. To maintain compatibility with older versions we can still use this only when non-whitespace characters are specified. Unfortunately no similar overloads exist for LTRIM/RTRIM (#22924).

Note that TRIM has a funky syntax (`TRIM ( [ characters FROM ] string )`), instead of just commas.

**Comment by maumar** [comment]
mysql has to deal with this syntax in some places, we could introduce a similar solution:

https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/blob/master/src/EFCore.MySql/Query/Expressions/Internal/MySqlComplexFunctionArgumentExpression.cs

**Comment by roji** [comment]
Yeah, [I have one too](https://github.com/npgsql/EFCore.PG/blob/35703a17f82150bcdb90b7e0e8b356e7b972c42c/src/EFCore.PG/Query/Expressions/Internal/PostgresFunctionExpression.cs) :rofl: Finally SQL Server will join the exclusive clique....

**Comment by abdielcs** [comment]
I would like to know what you expect the syntax of these new Trim functions to be. As far as I can see, the use of leading and trailing is the same as using TrimStart and TrimEnd. Are you interested in having overlapping functions like MyString.TrimLeading(trimChars) and MyString.TrimTrailing(trimChars) that generate a different SQL? Or do you expect something else? The only real addition could be MyString.TrimBoth(trimChars), but this could be achieved with the old Trim function with an argument.

**Comment by abdielcs** [comment]
I would like to know how you expect the new TRIM function with parameters to work. Could it be easily translated to something like LTRIM(RTRIM('MyString', 'CharsToTrim'), 'CharsToTrim'), or are you expecting to use the new TRIM ( [ characters FROM ] string ) syntax?

**Comment by roji** [comment]
We should definitely translate to the new SQL Server 2022 function, but the quirky syntax makes this a more difficult task. That goes beyond a beginner-level issue for sure.
